modules/mod_ginger_sword/mod_ginger_sword.erl

```diff
@@ -0,0 +1,232 @@
+
+update_rsc(RscId, Context) ->
+    URI = format_rsc_uri(RscId, Context),
+    Body = render_rdf_body(RscId, Context),
+    ?zInfo("Updating " ++ format_rsc_uri(RscId, Context) ++ " in SWORD endpoint", Context),
+    http_request(put, {URI, [], "application/json", Body}, Context).
+
+delete_rsc(RscId, Context) ->
+    %% Make sure no more create and update tasks are scheduled
+    z_pivot_rsc:delete_task(mod_ginger_sword, dispatch_task,
+                            unique_key(create, RscId), Context),
+    z_pivot_rsc:delete_task(mod_ginger_sword, dispatch_task,
+                            unique_key(update, RscId), Context),
+
+    URI = format_rsc_uri(RscId, Context),
+    ?zInfo("Deleting " ++ format_rsc_uri(RscId, Context) ++ " from SWORD endpoint", Context),
+    http_request(delete, {URI, []}, Context).
+
+%% Task queueing
+
+%% @doc Creates a unique key for referencing the task
+unique_key(Task, RscId) ->
+    "sword-" ++ z_convert:to_list(Task) ++ "-" ++ z_convert:to_list(RscId).
+
+%% @doc Checks if HTTP response contains error code
+resp_is_succes(Resp) ->
+    {_, {{_, Code, _}, _Headers, _Body}} = Resp,
+    (Code >= 200) and (Code < 300).
+
+%% @doc Dispatches a task from the queue
+%% If an error of some sort occurs delay the task
+dispatch_task(create, RscId, Context) ->
+    dispatch_task(fun create_rsc/2, RscId, Context);
+dispatch_task(update, RscId, Context) ->
+    dispatch_task(fun update_rsc/2, RscId, Context);
+dispatch_task(delete, RscId, Context) ->
+    dispatch_task(fun delete_rsc/2, RscId, Context);
+dispatch_task(TaskFn, RscId, Context) when is_function(TaskFn) ->
+    try
+        case TaskFn(RscId, Context) of
+            {error, _} ->
+                {delay, 60};
+            Resp ->
+                case resp_is_succes(Resp) of
+                    true -> Resp;
+                    false -> {delay, 60}
+                end
+        end
+    catch
+        Error:Reason ->
+            ?zWarning(io_lib:format("Sword publish failed(~p:~p):~n ~p",
+                                    [Error, Reason, erlang:get_stacktrace()]),
+                      Context),
+            {delay, 60}
+    end.
+
+%% @doc Cancels all scheduled publication tasks
+cancel_tasks(Context) ->
+    z_pivot_rsc:delete_task(mod_ginger_sword, dispatch_task, Context).
+
+%% @doc Adds a new durable task to the queue
+queue_task(Task, RscId, Context) ->
+    %% insert_task(Module, Function, UniqueKey, Args, Context)
+    z_pivot_rsc:insert_task(mod_ginger_sword,
+                            dispatch_task,
+                            unique_key(Task, RscId),
+                            [Task, RscId],
+                            Context).
+
+%% @doc Takes a category path and determines if it must be published
+cat_must_publish(IsA) ->
+    ExactCat = case IsA of
+                   [] ->
+                       undefined;
+                   _ ->
+                       lists:last(IsA)
+               end,
```

Comment:
I would replace this `case` with a pattern match: `cat_must_publish([])`. See also https://github.com/inaka/erlang_guidelines#avoid-unnecesary-calls-to-length1 (although you’re not using `length/1`, the example is similar).

Reply:
Check out the update, it's totally gone now :)